textile/features.textile

```diff
@@ -2349,6 +2361,9 @@ class ReferenceExtras: // REX*
   timeserial: String // REX2a
   type: String //REX2b
 
+class ClientLibraryInfo: // CLI*
```

Comment:
Do you think that it needs to link to other SDK classes? It's a class which only has static methods (i.e. you can use its functionality without needing an instance of it), so I thought people wanting to use it would just access the class from the global namespace.

Reply:
Oh interesting, I hadn't thought of that. I'm not sure how that would work when libraries need to gather information after initialisation to populate the agent string (eg ably-js needs to run `process.version()` to get the NodeJS version... maybe we could run that as an import side-effect but I would prefer not tbh).

Also, is the expectation that any agents configured via the `ClientOptions.agent` option would not be present in this list?

Reply:
> I'm not sure how that would work when libraries need to gather information after initialisation to populate the agent string (eg ably-js needs to run `process.version()` to get the NodeJS version

Perhaps I've not understood properly here, but is your concern to do with the fact that `ClientLibraryInfo.agents` is specified as a property here and hence you're worried that we wouldn't be able to run code to derive its value? If you were to think of it as a class method instead of a property would that solve this issue?

> Also, is the expectation that any agents configured via the `ClientOptions.agent` option would not be present in this list?

Correct. The `ClientOptions#agents` property is used to configure a specific _instance_ of `RestClient` or `RealtimeClient`. The intention is that the `ClientLibraryInfo` interface provides information about the library without needing to instantiate a client.

Reply:
Thanks for the explanation. Yeah, a method would work 👍 Although now that you've explained your intention of the word "default", would that also exclude stuff like environment/runtime versions? If so then we could probably get away with it just being a property.

Reply:
It does include environment/runtime versions, yep. I've added an example to CR2a to make this clear.

Reply:
awesome, that example looks good thanks

Reply:
I've added a bit about the choice to use a property in the commit message: https://github.com/ably/specification/pull/113/commits/4654b93e4237c70ac5457d4294f29ef291739c19